contracts/bridge/Bridge.sol

```diff
@@ -121,15 +122,16 @@ contract Bridge is IERC20BridgeReceiver, IERC721BridgeReceiver, BridgeFee, Bridg
         delete allowedTokens[_token];
     }
 
-    function updateHandleNonce(uint64 _requestedNonce) internal {
+    function updateHandleNonce(uint64 _requestedNonce, uint64 _requestedBlockNumber) internal {
         uint64 i;
-        handledNonces[_requestedNonce] = true;
+        handledHistory[_requestedNonce] = _requestedBlockNumber;
 
         if (_requestedNonce > maxHandledRequestedNonce) {
             maxHandledRequestedNonce = _requestedNonce;
         }
-        for (i = sequentialHandleNonce; i <= maxHandledRequestedNonce && handledNonces[i]; i++) { }
+        for (i = sequentialHandleNonce; i <= maxHandledRequestedNonce && handledHistory[i] > 0; i++) { }
```

Comment:
I
```suggestion
        for (i = sequentialHandleNonce; i <= maxHandledRequestedNonce; i++) { }
```
I think it is okay to remove the condition.

Reply:
To recover correctly, we need to keep the maximum sequential handle nonce history because arbitrary nonces can be quested after #80. Checking `handledHistory[i] > 0` is for the sequential update.

Reply:
Thank you for the explanation!